config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java

```diff
@@ -1121,9 +1122,21 @@ String buildLegacyOptions() {
             return jvmOptions;
         }
 
-        private void log(String jvmOptions) {
-            if (isHosted && jvmOptions != null && !jvmOptions.isEmpty())
-                logger.logApplicationPackage(Level.INFO, "JVM options from services.xml: " + jvmOptions);
+        private void validateJvmOptions(String jvmOptions) {
+            if (jvmOptions == null || jvmOptions.isEmpty()) return;
+
+            String[] optionList = jvmOptions.split(" ");
+            List<String> invalidOptions = Arrays.stream(optionList)
+                                                .filter(option -> !option.isEmpty())
+                                                .filter(option -> !Pattern.matches(validPattern.pattern(), option))
+                                                .sorted()
+                                                .collect(Collectors.toList());
+
+            String message = "Invalid JVM options in services.xml: " + String.join(",", invalidOptions);
```

Comment:
I guess you should return here if invalidOptions is empty. If not you will throw or log an empty message.

Reply:
Right, thanks, will fix